src/file_dialog.rs

```diff
@@ -180,6 +180,12 @@ impl Debug for dyn FileDialogModal + Send + Sync {
     }
 }
 
+/// Callback type to inject a custom egui ui inside the file dialog's ui.
+///
+/// Also gives access to the file dialog, since it would otherwise be inaccessible
+/// inside the closure.
+type FileDialogUiCallback<'a> = dyn FnMut(&mut egui::Ui, &mut FileDialog) + 'a;
```

Comment:
I added the `<'a>` lifetime here because otherwise there seemed to be lifetime issues when trying to borrow things outside of the closure in my testing code.

Reply:
Sounds good 👍